app/models/fe/question.rb

```diff
@@ -186,7 +186,7 @@ def set_response(values, answer_sheet)
         #   raise object_name.inspect + ' == ' + attribute_name.inspect
         # end
       else
-        @answers ||= []
+        @answers ||= sheet_answers.to_a
```

Comment:
@twinge this is what stops the empty string answers.  Note that I moved this from the question_set line "-        question.answers = question.responses(answer_sheet) #answers[question.id]" into here.  Seemed to make more sense to me that a question can set its own current answers list.  The switch that fixes the empty strings though is going from question.responses(answer_sheet) to sheet_answers.  Responses method filters out empty string answers.  Once the answers array has the empty string answer properly, the rest of the method handles the cases correctly -- if a new answer is given, the empty string answer is marked for deletion, if an empty string answer is given, it recognizes there's already an answer with the empty string and does nothing.